rafamadriz/friendly-snippets
# feat(typescript): tsdoc snippets

This PR implements the full [tsdoc specification](https://typedoc.org/tags/example/) with its original descriptions, providing a great user experience. 

## Demo
![screenshot_2023-05-30_20-49-07_243766568](https://github.com/rafamadriz/friendly-snippets/assets/3357792/5a1c5f74-6cc8-45d2-a0a1-dd023ed26983)
![screenshot_2023-05-30_20-55-51_706605122](https://github.com/rafamadriz/friendly-snippets/assets/3357792/837b5040-d4b8-4bde-8e53-37071c7f721c)
![screenshot_2023-05-30_20-07-58_125737314](https://github.com/rafamadriz/friendly-snippets/assets/3357792/83cd7242-178c-432a-9f49-d28c85820012)

![screenshot_2023-05-30_20-06-33_078358421](https://github.com/rafamadriz/friendly-snippets/assets/3357792/5b12e7ae-7a72-4c50-a81d-7233738b00c6)

## What implements
* All the 37 reserved keywords for tsdoc comments.
* 2 extra snippets for /* and /** to create simple, or full comments respectively. These are the only two with guided auto completion.

## How to test

Clone the project from Zeioth/friendly-snippets and add this to your LuaSnip config

``` lua
-- Enable friently-snippets opt in snippets
require("luasnip").filetype_extend("typescript", { "tsdoc" })
```

Now open any .ts file and the snippets should be available.
